fetchcode/giturl.py

```diff
@@ -0,0 +1,54 @@
+# fetchcode is a free software tool from nexB Inc. and others.
+# Visit https://github.com/nexB/fetchcode for support and download.
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# http://nexb.com and http://aboutcode.org
+#
+# This software is licensed under the Apache License version 2.0.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: 
+# http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import os
+import subprocess
+
+from urllib.parse import urlparse, urlunparse
+
+
+def can_handle(source):
+	url_parts = urlparse(source)
+	if url_parts.scheme not in ('http', 'https', 'git', ''):
+		return False
+	elif not url_parts.scheme:
+		return os.path.exists(os.path.join(source, '.git'))
+	else:
+		return True
+
+
+def clone(source, dest, branch='master', depth=None):
+	if not can_handle(source):
+	    raise UnhandledSource('Cannot handle {}'.format(source))
+	if os.path.exists(dest):
```

Comment:
I am not sure we would want to do a pull if a path existed. Instead this should be an error